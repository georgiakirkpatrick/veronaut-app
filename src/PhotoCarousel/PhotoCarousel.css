.PhotoCarousel {
    width: 100%;
    overflow: scroll;
}

.PhotoCarousel__items {
    display: flex;
    flex-direction: row;
    overflow: scroll;
}

.PhotoCarousel__items > li > img {
    width: 80vw;
    padding: 0 10px 0 0;
}

.PhotoCarousel__items > div {
    padding: 0 10px 0 0;
    width: 300px;
}

#section-carousel > .PhotoCarousel__items {
    width: 150%;
}