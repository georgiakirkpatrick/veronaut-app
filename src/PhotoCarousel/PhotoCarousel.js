import React from 'react'
import './PhotoCarousel.css'

const PhotoCarousel = props => {
    const {
        colorArray,
        id,
        productName,
        selectedColorId
    } = props
 
    const makeColorImages = color => {
        const primaryImage = color.imageArray.find(image => image.primaryImage === true)
        const allButPrimary = color.imageArray.filter(image => image.imageId !== primaryImage.imageId)

        return [
            primaryImage,
            ...allButPrimary
        ]
    }

    const selectedColor = colorArray.filter(color => color.colorId === selectedColorId)[0]
    const imagesForColor = makeColorImages(selectedColor)
    const formattedImages = imagesForColor.map(image => {

        return <li key={'li-' + image.imageId}>
            <img
                key={'image-' + image.imageId}
                src={image.imageUrl}
                alt={`${productName} ${selectedColor.colorEngName} ${image.imageId}`}
            />
        </li>
    })

    return (
        <div 
            id={id}
            className='PhotoCarousel'
        >
            <ul className='PhotoCarousel__items'>
                {formattedImages}
            </ul>
        </div>
    )
}

PhotoCarousel.defaultProps = {
    id: 0,
    colorArray: [
        {
            color_description_id: 1,
            color_english_name: '',
            id: 1,
            image_array: [
                {
                    id: 1,
                    image_url: '',
                    primary_image_for_color: true,
                }
            ],
            swatch_image_url: ''
        }
    ],
    selectedColorId: 1,
    productName: ''
}

export default PhotoCarousel