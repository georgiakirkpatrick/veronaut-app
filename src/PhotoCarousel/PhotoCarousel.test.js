import React from 'react'
import ReactDOM from 'react-dom'
import PhotoCarousel from './PhotoCarousel'
import renderer from 'react-test-renderer'

describe ('PhotoCarousel', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<PhotoCarousel />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<PhotoCarousel />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})