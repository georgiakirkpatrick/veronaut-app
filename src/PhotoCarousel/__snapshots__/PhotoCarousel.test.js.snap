// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoCarousel renders the UI as expected 1`] = `
<div
  className="PhotoCarousel"
  id={0}
>
  <ul
    className="PhotoCarousel__items"
  >
    <li>
      <img
        alt="  1"
        src=""
      />
    </li>
  </ul>
</div>
`;
